e2e-tests/development-runtime/cypress/integration/unified-routing/collection-routing.js

```diff
@@ -0,0 +1,22 @@
+describe(`collection-routing`, () => {
+  beforeEach(() => {
+    cy.visit(`/`).waitForRouteChange()
+  })
+
+  it(`can navigate to a collection route and see its content rendered`, () => {
+    cy.visit(`/collection-routing/root`).waitForRouteChange()
+
+    cy.getTestElement(`collection-routing-blog`)
+      .invoke(`attr`, `data-testslug`)
+      .then(slug => {
+        // should navigate us to an actual collection builder route.
+        cy.getTestElement(`collection-routing-blog`)
+          .first()
+          .click()
+          .waitForRouteChange()
+          .getTestElement(`slug`)
+          .invoke(`text`)
+          .should(`equal`, slug)
+      })
+  })
```

Comment:
Do we want to add a few more test cases here? Like making sure all blog posts are rendered, ...

Reply:
Making sure the blog posts rendered is inferred. This test will crash if that doesn't work.

But added a test to ensure Unions work and that our collectionQuery api works too!